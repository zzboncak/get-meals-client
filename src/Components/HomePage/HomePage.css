.app-description {
    width: 95%;
    margin-left: auto;
    margin-right: auto;
    font-family: 'Ubuntu';
}

.hero-header {
    font-family: 'Kreon';
}

.home-page__section {
    display: flex;
    flex-direction: column;
    width: 95%;
    margin-left: auto;
    margin-right: auto;
    justify-content: space-between;
}

.map {
    width: 75%;
}

img.map {
    width: 100%;
    overflow: hidden;
}

@media screen and (min-width: 400px) {
    .home-page__section {
        flex-direction: row;
        justify-content: space-between;
    }
}