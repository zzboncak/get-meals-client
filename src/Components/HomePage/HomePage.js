import React from 'react';
import Sidebar from '../Sidebar/Sidebar';
import Popup from '../Popup/Popup';
import './HomePage.css';
import mapSample from '../Util/map.png';

class HomePage extends React.Component {
    render() {
        return (
            <div>
                <section className="main-view">
                    <Sidebar />
                    <div className="map">
                        <img src={mapSample} alt="map sample"/>
                    </div>
                </section>
            </div>
        )
    }
}

export default HomePage;