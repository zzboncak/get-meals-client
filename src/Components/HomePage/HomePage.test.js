import React from 'react';
import ReactDOM from 'react-dom';
import HomePage from './HomePage';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe(`HomePage component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<HomePage />, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <HomePage />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
