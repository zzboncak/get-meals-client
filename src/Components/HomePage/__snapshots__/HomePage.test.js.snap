// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders the component as expected 1`] = `
<div>
  <section
    className="app-description"
  >
    <h1
      className="hero-header"
    >
      Welcome to Get Meals
    </h1>
    <p>
      A platform to find and share locations serving food and meals to those in need.
    </p>
    <p>
      Enter your location on the left to find places serving free food near you. Or, click on the "Add Location" in the nav bar above to report a place that you know is serving free food.
    </p>
    <p>
      Let's all do our part in letting no one go hungry.
    </p>
  </section>
  <section
    className="main-view home-page__section"
  >
    <div
      className="sidebar-container"
    >
      <form
        className="sidebar-form"
        onSubmit={[Function]}
      >
        <div
          className="sidebar-form__search"
        >
          <label
            htmlFor="sidebar-form__label"
          >
            Search by address, city, or zip code:
          </label>
          <input
            className="sidebar-form__input"
            id="address"
            placeholder="New York, NY"
            required={true}
            type="text"
          />
        </div>
        <div
          className="sidebar-form__button-container"
        >
          <button
            className="sidebar-form__submit-button"
            type="submit"
          >
             Search
          </button>
        </div>
      </form>
    </div>
    <div
      className="map"
    >
      <div
        className="mapContainer"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </div>
    </div>
  </section>
</div>
`;
